scipy/sparse/linalg/_interface.py

```diff
@@ -590,7 +590,11 @@ def _matmat(self, X):
             return super()._matmat(X)
 
     def _matvec(self, x):
-        return self.__matvec_impl(x)
+        if self.__matvec_impl is not None:
+            return self.__matvec_impl(x)
+        else:
+            raise TypeError("Tthis LinearOperator doesn't implement `rmatvec` nor `adjoint`,"
```

Comment:
There's a small typo here
```suggestion
            raise TypeError("This LinearOperator doesn't implement `rmatvec` nor `adjoint`,"
```